### PR TITLE
sandbox.promise → sandbox.waitForLoad()

### PR DESCRIPTION
**Pull request by hjdivad**

### CODE REVIEW

**Comment by hjdivad** [comment]
cc @tomdale

**Comment by stefanpenner** [comment]
Seems reasonable, although passing around defer's as locks gets cumbersome,  and makes some of the code paths hard to follow.

We used the new api in our [ghetto-spike](https://github.com/stefanpenner/oasis.js/compare/inline) for an inline adapter

**Comment by hjdivad** [comment]
@stefanpenner have a suggested alternative for the following pattern?

``` js
a.waitForEvent().then(doSomething)
a.doEvent()
```

It would seem weird to get around this by eg having `waitForload()` implicitly load, or by having to call `load()` when the semantics you're looking for are `waitForLoad`.
